src/nbl/asset/utils/CSPIRVIntrospector.cpp

```diff
@@ -109,20 +109,127 @@ static CSPIRVIntrospector::CStageIntrospectionData::VAR_TYPE spvcrossType2E_TYPE
 // returns true if successfully added all the info to self, false if incompatible with what's already in our pipeline or incomplete (e.g. missing spec constants)
 NBL_API2 bool CSPIRVIntrospector::CPipelineIntrospectionData::merge(const CSPIRVIntrospector::CStageIntrospectionData* stageData, const ICPUShader::SSpecInfoBase::spec_constant_map_t* specConstants)
 {
-    return false;
+
+    // merge descriptors
+    for (uint32_t i = 0u; i < ICPUPipelineLayout::DESCRIPTOR_SET_COUNT; i++)
+    {
+        std::ostringstream debug;
+        debug << "ds ID: " << i << std::endl;
+        const auto& bindingInfos = stageData->getDescriptorSetInfo(i);
+        for (const auto& bindingInfo : bindingInfos)
+        {
+            CPipelineIntrospectionData::SDescriptorInfo descInfo;
+            descInfo.binding = bindingInfo.binding;
+            descInfo.type = bindingInfo.type;
+            
+            const auto& foundBinding = m_descriptorSetBindings[i].find(descInfo);
+            if (foundBinding != m_descriptorSetBindings[i].end())
+            {
+                // TODO: validation
```

Comment:
just check that the:
- type
- stride
- count
are matching.

An added condition (regardless of if you find a binding with same number) is that if your `descInfo.isRuntimeSized()` then either:
- `descInfo` has a higher or binding number than anything in `m_descriptorSetBindings[i]` (might pay off to have a `m_highestBindingNumber[4]` in the pipeline intro)
- `descInfo` is equal to `m_highestBindingNumber[i]` both need to be runtime sized -> but then you go into the `foundBinding != m_descriptorSetBindings[i].end()` case so the counts will be checked
- `m_descriptorSetBindings.empty()` 


overall a simple check of
```cpp
if (bindingInfo.count.isRuntimeSized() && bindingInfo.binding<m_highestBindingNumber[i])
    return false;
```
will do